include/zephyr/drivers/stepper.h

```diff
@@ -235,105 +162,10 @@ __subsystem struct stepper_driver_api {
- *
- * @retval -EIO Error during Enabling
- * @retval 0 Success
- */
-__syscall int stepper_enable(const struct device *dev);
-
-static inline int z_impl_stepper_enable(const struct device *dev)
-{
-	const struct stepper_driver_api *api = (const struct stepper_driver_api *)dev->api;
-
-	return api->enable(dev);
-}
-
-/**
- * @brief Disable stepper driver
- *
- * @details Disabling the driver shall switch off the power stage and de-energize the coils.
- * Disabling the stepper does not implicitly stop the stepper. If the motor shall not move after
- * re-enabling the stepper than consider calling stepper_stop() before.
- *
- * @param dev pointer to the stepper driver instance
- *
- * @retval  -ENOTSUP Disabling of driver is not supported.
- * @retval -EIO Error during Disabling
- * @retval 0 Success
- */
-__syscall int stepper_disable(const struct device *dev);
-
-static inline int z_impl_stepper_disable(const struct device *dev)
-{
-	const struct stepper_driver_api *api = (const struct stepper_driver_api *)dev->api;
-
-	return api->disable(dev);
-}
-
-/**
- * @brief Set the micro-step resolution in stepper driver
- *
- * @param dev pointer to the stepper driver instance
- * @param resolution micro-step resolution
- *
- * @retval -EIO General input / output error
- * @retval -ENOSYS If not implemented by device driver
- * @retval -EINVAL If the requested resolution is invalid
- * @retval -ENOTSUP If the requested resolution is not supported
- * @retval 0 Success
- */
-__syscall int stepper_set_micro_step_res(const struct device *dev,
-					 enum stepper_micro_step_resolution resolution);
-
-static inline int z_impl_stepper_set_micro_step_res(const struct device *dev,
-						    enum stepper_micro_step_resolution resolution)
-{
-	const struct stepper_driver_api *api = (const struct stepper_driver_api *)dev->api;
-
-	if (api->set_micro_step_res == NULL) {
-		return -ENOSYS;
-	}
-
-	if (!VALID_MICRO_STEP_RES(resolution)) {
-		return -EINVAL;
-	}
-	return api->set_micro_step_res(dev, resolution);
-}
-
-/**
- * @brief Get the micro-step resolution in stepper driver
- *
- * @param dev pointer to the stepper driver instance
- * @param resolution micro-step resolution
- *
- * @retval -EIO General input / output error
- * @retval -ENOSYS If not implemented by device driver
- * @retval 0 Success
- */
-__syscall int stepper_get_micro_step_res(const struct device *dev,
-					 enum stepper_micro_step_resolution *resolution);
-
-static inline int z_impl_stepper_get_micro_step_res(const struct device *dev,
-						    enum stepper_micro_step_resolution *resolution)
-{
-	const struct stepper_driver_api *api = (const struct stepper_driver_api *)dev->api;
-
-	if (api->get_micro_step_res == NULL) {
-		return -ENOSYS;
-	}
-	return api->get_micro_step_res(dev, resolution);
-}
-
 /**
  * @brief Set the reference position of the stepper
  *
- * @param dev Pointer to the stepper driver instance.
+ * @param dev Pointer to the stepper controller instance.
```

Comment:
```suggestion
 * @param dev Pointer to the stepper instance.
```
Not sure what is the best decision in this (and the other similar cases), but this way it is consistent with the rest of the changes

Reply:
I don't consider this resolved with the latest changes, because while `Pointer to the stepper driver instance.` is technically correct, it causes confusion with the stepper_drv api, which is for stepper drivers. One option would be to reword the sentence, while the other one would be to use a sentence based on the other apis, e.g. `Pointer to the device structure for the driver instance.` or `Stepper device instance.`

Otherwise, everything else looks good to me.